### PR TITLE
Add support for ghc-8.10.4 and drop it for ghc-8.10.1

### LINKED ISSUE #1317: Add ghc-8.10.4 support
I was not aware of this one!

### PR DESCRIPTION
**Pull request by jneira**
* We dont have time and resources to support all possible ghc versions (as commented [here](https://github.com/haskell/haskell-language-server/issues/1317#issuecomment-774717281)) so, as we are supporting 8.8.2..8.8.4 i propose only support the last three minor version of each major version too for 8.10
* This pr drops the unused (afaik) ci azure config 
* Closes #1317

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -35,9 +35,9 @@ let
         ourHaskell = pkgs.haskell // {
             packages = pkgs.haskell.packages // {
                 # relax upper bounds on ghc 8.10.x versions (and skip running tests)
-                ghc8101 = extended (pkgs.haskell.packages.ghc8101.override sharedOverrides);
                 ghc8102 = extended (pkgs.haskell.packages.ghc8102.override sharedOverrides);
                 ghc8103 = extended (pkgs.haskell.packages.ghc8103.override sharedOverrides);
+                ghc8104 = extended (pkgs.haskell.packages.ghc8104.override sharedOverrides);
```
Comment (jneira) [inline nix/default.nix:40]: Not sure it this will work out of the box

**Comment by jneira** [comment]
What do you think in make a github release after adding ghc-8.10.4 support?, it would have hiedb included
//cc @pepeiborra @wz1000

**Comment by jneira** [comment]
When ci is green i will remove the github required checks for ghc-8.10.1 and add ghc-8.10.4 ones

**Review by Ailrun — APPROVED** [review]
Awesome, Thanks. I hope there's no one left behind with GHC 8.10.1...

**Comment by wz1000** [comment]
I'm OK with a release.

There have been no bug reports about hiedb so far, so either its working as intended (yay!), or nobody is using it. If its the latter, I'm ready for people to start dropping the bug reports once they start using it.

**Comment by jneira** [comment]
I need some window time to 

1. disable actual ci checks for the test suite for 8.10.3
2. merge the pr
3. enable new ci checks for the test suite and 8.10.4

and no merge other pr's with a failing test suite between 1 and 2

Then we should update the rest of pr's (that is enforced by github)

**Comment by Ailrun** [comment]
I think we can do 1. and 3. together. If so, all other PRs cannot be merged until this one is merged.

**Comment by jneira** [comment]
yeah, I didn't want block anyone though, I will wait until I come home and will change them if I can't do the switch &merge fast

**Pull request merged**